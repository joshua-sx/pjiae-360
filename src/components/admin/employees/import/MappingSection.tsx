import { EmployeeColumnMapping } from './EmployeeColumnMapping';
import { EmployeeImportData } from './types';

interface MappingSectionProps {
  uploadMethod: 'upload' | 'paste' | 'manual';
  headers: string[];
  rows: string[][];
  columnMapping: Record<string, string>;
  uploadedFile: { name: string; size: number } | null;
  onComplete: (data: EmployeeImportData) => void;
  onBack: () => void;
}

export function MappingSection({
  uploadMethod,
  headers,
  rows,
  columnMapping,
  uploadedFile,
  onComplete,
  onBack,
}: MappingSectionProps) {
  return (
    <EmployeeColumnMapping
      data={{
        uploadMethod,
        csvData: {
          rawData: '',
          headers,
          rows,
          columnMapping,
        },
        uploadedFile,
        manualEmployees: [],
      }}
      onDataChange={() => {}}
      onNext={onComplete}
      onBack={onBack}
    />
  );
}

