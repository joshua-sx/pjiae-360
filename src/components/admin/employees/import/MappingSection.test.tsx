import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import { MappingSection } from './MappingSection';

describe('MappingSection', () => {
  it('renders consistently', () => {
    const { container } = render(
      <MappingSection
        uploadMethod="upload"
        headers={[]}
        rows={[]}
        columnMapping={{}}
        uploadedFile={null}
        onComplete={() => {}}
        onBack={() => {}}
      />
    );
    expect(container).toMatchSnapshot();
  });
});

