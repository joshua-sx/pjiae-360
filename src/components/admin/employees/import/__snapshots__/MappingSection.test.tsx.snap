// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MappingSection > renders consistently 1`] = `
<div>
  <div
    class="space-y-6"
  >
    <div
      class="flex items-center gap-4"
    >
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:shrink-0 touch-manipulation border border-input bg-background hover:bg-accent hover:text-accent-foreground h-8 px-3 text-xs min-h-[44px] sm:min-h-0 [&_svg]:size-3"
      >
        <svg
          class="lucide lucide-arrow-left mr-2 h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m12 19-7-7 7-7"
          />
          <path
            d="M19 12H5"
          />
        </svg>
        Back
      </button>
      <div>
        <h1
          class="text-2xl font-bold"
        >
          Column Mapping
        </h1>
        <p
          class="text-muted-foreground"
        >
          Map your CSV columns to employee fields
        </p>
      </div>
    </div>
    <div
      class="flex items-center justify-center min-h-96"
    >
      <div
        class="rounded-lg border bg-card text-card-foreground shadow-sm max-w-md border-orange-200"
      >
        <div
          class="sm:p-6 p-6 text-center"
        >
          <svg
            class="lucide lucide-circle-alert w-12 h-12 text-orange-500 mx-auto mb-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="12"
              x2="12"
              y1="8"
              y2="12"
            />
            <line
              x1="12"
              x2="12.01"
              y1="16"
              y2="16"
            />
          </svg>
          <h3
            class="text-lg font-semibold mb-2"
          >
            No CSV Data Found
          </h3>
          <p
            class="text-muted-foreground mb-4"
          >
            Please go back and upload a CSV file first.
          </p>
          <button
            class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:shrink-0 touch-manipulation border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 text-sm min-h-[44px] sm:min-h-0 [&_svg]:size-4"
          >
            Go Back
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
